src/ucp/core/ucp_ep.c

```diff
@@ -706,6 +706,44 @@ ucs_status_t ucp_ep_create(ucp_worker_h worker, const ucp_ep_params_t *params,
     return status;
 }
 
+ucs_status_t ucp_ep_query(ucp_ep_h ep, ucp_ep_attr_t *attr)
+{
+    uct_ep_h uct_cm_ep = ucp_ep_get_cm_uct_ep(ep);
+    uct_ep_attr_t uct_cm_ep_attr;
+    ucs_status_t status;
+
+    if (uct_cm_ep == NULL) {
```

Comment:
should fail only if field_mask has UCP_EP_ATTR_FIELD_LOCAL_SOCKADDR or UCP_EP_ATTR_FIELD_REMOTE_SOCKADDR